File: elixir/prime_factors/lib/pf/prime_factors.ex
defmodule PF.PrimeFactors do
  def prime_factors(num, candidate \\ 2) do
    case num do
        1 -> []
        x when rem(x, candidate) == 0 -> [candidate|prime_factors(div(x, candidate))]
        x -> prime_factors(x, candidate + 1)
    end
  end
end

File: haskell/PrimeFactors.hs
module PrimeFactors (primeFactors) where

primeFactors :: Integer -> [Integer]

primeFactors x = primeFactors x 2
    where
        primeFactors x candidate
            | x == 1 = []
            | x `mod` candidate == 0 = [candidate] ++ primeFactors (x `div` candidate) candidate
            | otherwise = primeFactors x (candidate + 1)
